Loc key collisions

Harain opened the pull request:
just a draft for now, a bit over halfway done

Harain committed:
loc key collisons fix part 1

this is to clear entries in text.log - loc key collisions (duplicates) - cleared 711 lines, 3334 to go (omg)
all types of these:
- repeated on our side (sometimes in same file, sometimes in different files)
- whole focus trees duplicated (PRU_l_english.yml and parts of RAJ_l_english.yml)
- second name instead of name and desc
- vanilla repeated
- vanilla replaced but outside replace folder
- probably other cases i can't remember while writing this

stuff that just repeats itself I simply deleted
exact copies of vanilla I simply deleted
vanilla replaced outside of replace folder I moved to new file r56_replace_l_english.yml
whenever in doubt which of our duplicated entries to remove, I chose to keep the newer version...

Harain committed:
Update GRE_l_english.yml

- commented lines in bftb_ideas_l_english.yml that are in this file also
- lines moved to replace marked with a comment
- commented lines repeated in same file
- deleted GRE_GATHERING_STORM_DESC (same desc as vanilla) repeated in r56_bookmarks_l_english.yml
- deleted royal_hellenic_naval_shipyard: same as vanilla
3292 lines to go -.-

Harain committed:
Fixed game_rules_l_english.yml loc duplicates

just deleted repeated lines, keeping the ones that differ from vanilla
3272 lines to go

Harain committed:
fixed GER_l_english.yml loc duplicates

deleted vanilla exact copies (some are in copied over files, which I left there, some not)
commented lines in vanilla files wtt_decisions_l_english.yml and focus_l_english.yml repeated here
moved lines to replace (left them commented)
3226 to go

Harain committed:
fix NOR_l_english.yml duplicate locs

deleted locs duplicated in another r56 file (r56_countries_cosmetic_l_english.yml, fixed a loc here)
comented exact vanilla duplicates

Harain committed:
fixed ALB_l_english.yml duplicate locs

deleted exact vanilla copies
moved modified vanilla locs to replace
3150 lines to go...

Harain committed:
fix YUG_l_english.yml loc duplicates

deleted exact vanilla copies
moved modified vanilla locs to replace
duplicates of modified vanilla locs commented in copied over vanilla file (dod_focus_l_english.yml)
deleted duplicate in same file (exact copy)
commented duplicate in same file (different)

Harain committed:
Merge branch 'waffle-january-lastminutefixes' into loc-key-collisions

Harain committed:
fix countries_l_english.yml duplicate locs

we have copied the vanilla file countries_l_english.yml over
that in itself wouldn't be too bad, the problem is lack of consistency in working with it:
some ppl edit it directly (without commenting), others just add the locs to an r56 file, while leaving the vanilla loc there uncommented

what I did was to comment (in countries_l_english.yml) those we use in an r56 file
deleted most of those that are exact vanilla copies and moved different duplicate to replace
tried to be as thorough as possible, but that file is a mess, so can't be absolutely sure I didn't make any mistake
when in doubt I chose the version that was different from vanilla, hope that was always the correct choice

Also, even after moving them to replace, 6 locs still throw a duplicate error, so added a new replace file (r56_replace2_l_english.yml) to keep them together for further investigation later

this cleared 1066 lines from text.log, 2062 to go

PS: there might still be exact vanilla copies in replace, but since it's not a big issue, and it's a LOT of work to go through them all again, I decided not to

Harain committed:
fix r56_countries_cosmetic_l_english.yml duplicates

Harain committed:
moved parties_l_english.yml

it doesn't overide the whole file if in replace folder, only individual loc keys

Harain committed:
delete interface/r56_parties_l_english.yml

seems like an older version copied into interface by mistake, comparing with the one we have in replace folder
just moved a loc in replace/r56_parties_l_english.yml to its proper place

Harain committed:
fix duplicate parties loc keys

commented in vanilla parties_l_english.yml those we have somewhere else
besides parties also also fixed some event locs in TIB_l_english.yml that were duplicated and needed to make 2 minor changes to events/Tibet.txt
made some party fixes in the other files in this commit, between duplicated in several files or in the file itself, and some minor fixes
when in doubt tried to pick the one that felt better

Harain committed:
update .gitignore

Harain committed:
fix LUX_l_english.yml duplicate locs

mostly exact copies from vanilla which I just deleted, except one, which I left commented near our custom desc
when duplicated in same file, chose the most recent, or deleted if exact same

Harain committed:
fix LIT_l_english.yml duplicates

some moved to replace
some duplicated in LIT_l_english.yml commented in copied vamilla file nsb_ideas_l_english.yml
deleted some exact vanilla copies
added leading space to all lines missing it

Harain committed:
fix r56e_equip_motor_mech_l_english.yml duplicate locs

some were duplicated in copied vanilla file equipment_l_english.yml, commented there
rest were all mistakes in loc keys

Harain committed:
fix ICE_l_english.yml duplicate locs

deleted exact vanilla copies and moved modified to replace

Harain committed:
fix SWE_l_english.yml loc duplicates

some moved to replace
some duplicated in here commented in copied vamilla file aat_focus_l_english.yml
deleted some exact vanilla copies
added leading space to all lines missing it

marijn211 committed:
Merge branch 'Beta-Edits' into loc-key-collisions

Harain committed:
fix loc duplicates in 2 files

deleted exact vanilla copies and moved modified to replace

Harain committed:
fix BEL_l_english.yml loc duplicates

deleted exact vanilla copies and moved modified to replace
left some exact vanilla copies commented when next to our custom desc
also commented a loc in mtg_focus_l_english.yml that's repeated here

Harain committed:
fix my mistake

mass replace to add leading space turned those lit_ to uppercase by mistake, creatling a missing loc

Harain committed:
fix CZE_l_english.yml duplicate locs

moved exact vanilla copies to replace and removed extra leading spaces

Harain committed:
fix CHL_l_english.yml duplicate locs

moved 1 to replace, deleted exact vanilla copies and commented exact vanilla copies here in copied over vanilla file JUNO_focus_l_english.yml

Harain committed:
fix r56e_equip_tank_l_english.yml duplicate locs

the other 2 files (commented) had locs duplicated in r56e_equip_tank_l_english.yml

most duplicates are typos in loc keys, some easy to figure out, some not so much
list of doubts:
I assumed "Sexton" comes before "Sexton II" (even though a duplicate was the other way around)
light_tank_destroyer_equipment_2 for UKR, DDR, SOV, MON and TAN : "SU-37" or "SU-45"
heavy_tank_artillery_equipment_3 for ENG, NIR, SCO and WLS: "Black Prince AVRE" or "Ardeer Aggie" (commented the Ardeer Aggie, but it's probably a different kind of equipment)

Harain committed:
forgot these -.-

Harain committed:
more r56e_equip_tank_l_english.yml loc fixes

think I figured out the "Black Prince AVRE" vs "Ardeer Aggie" thing:
6 other countries only had "Ardeer Aggie" as WLS_heavy_tank_artillery_equipment_3, so am assuming that's the correct one
I also forgot to comment them in the last commit :p

Harain committed:
Merge branch 'waffle-january-lastminutefixes' into loc-key-collisions

Harain committed:
fix r56_companies_traits_l_english.yml duplicate locs

moved a bunch to replace
deleted 3 same as vanilla
deleted 1 in r56_traits_l_english.yml already here (moved to replace)

Harain committed:
misc loc duplicate fixes

Harain committed:
fix r56e_equip_aircraft_l_english.yml duplicate locs

just typos in loc keys, plus some duplicated here
commented weird one cause it was exact copy of vanilla and was throwing a duplicate error despite being in replace...

Harain committed:
fix duplicate locs

lots of files this time, sorting by vanilla file duplicated, making this faster for me:
deleted exact vanilla copies (unless next to a desc or related in which case left it commented)
moved modified vanilla locs to replace
mostly character names with some manufacturers in the mix, plus some misc stuff
954 lines to go, less than 1/4 now /scream

Harain committed:
fix TUR_l_english.yml duplicate locs

deleted exact vanilla copies, moved modified to replace

Harain committed:
fix HOL_l_english.yml duplicate locs

deleted exact vanilla copies, moved modified to replace
commented locs in mtg_focus_l_english.yml modified here
fixed a missing _desc in a duplicate loc key
commented out a large block of vanilla ship locs and left a note there

Harain committed:
fix URG_l_english.yml duplicate locs

deleted some repeated in the file
fixed a duplicate that had a missing _desc
commented entries in JUNO_focus_l_english.yml modified here

Harain committed:
fix more duplicate names

bunch of files, most just deleted exact vanilla copies, a couple modified moved to replace

Harain committed:
misc duplicate loc fixes

deleted exact vanilla copies, moved modified to replace
fixed a loc I broke earlier (only copied half to replace...)
fixed a loc in r56_prospecting_decisions_l_english.yml (was repeated there)

Harain committed:
fix equipment_l_english.yml duplicate locs

we have both equipment_l_english.yml and equip_air_l_english.yml copied over from vanilla with locs duplicated in both
kept the one in the original vanilla file and deleted the other (in equip_air_l_english.yml)
put cat_air_engine in replace since it doesn't belong in either file

we are also missing a large number of vanilla locs
vanilla equipment_l_english.yml has 4240 lines and ours only has 790 lines
vanilla equip_air_l_english.yml has 2028 lines and ours only has 40 lines
meaning we're missing over 5400 vamilla locs

Harain committed:
fix FRA_l_english.yml duplicate locs

same as previous commits, bla bla
am starting to lose it lol

Harain committed:
fix r56e_equip_artillery_l_english.yml duplicate locs

some typos, some duplicated somewhere else
note that many in this file are duplicated but NOT triggering duplicate error for some reason
didn't bother with those

Harain committed:
misc loc duplicate fixes

deleted duplicates in same file, a few fixes (added _desc to duplicate)
in BRA_l_english.yml there were duplicated manufacturers in english and portuguese, deleted the english ones
commented locs in vanilla files JUNO_focus_l_english.yml and lar_focus_l_english.yml that we have in r56 files

Harain committed:
fix focus_l_english.yml duplicate locs

deleted duplicates or commented vanilla duplicates when duplicates in 2 files (picked the one different from vanilla)
commented our locs that use §YMEFO bills§! instead of the vanila §Y[GetMilitaryProductionModifierName]§! which is better, since it account for bot types

Harain committed:
fix duplicate locs

nothing unusual

Harain committed:
fix duplicate locs

only thing unusual was 3 duplicate locs with totally different text for targeted_attack_theater_outcome in r56_operations_l_english.yml, commented the last 2

Harain committed:
Merge branch 'waffle-january-lastminutefixes' into loc-key-collisions

Harain committed:
more duplicate locs fixes

tons of the usual stuff
had to changes an event file to match new locs also

Harain committed:
last (!) duplicates loc fixes

there are still 8 duplicates I can't figure out, moved them all to a separate file to better keep track
localisation/english/replace/r56_replace2_l_english.yml

Harain committed:
remove vanilla _LaR_units_infantry.asset

added version to our copy of POR_infantry_entity
still getting error [18:08:46][no_game_date][pdx_entity.cpp:2171]: Duplicate of POR_infantry_entity added to entity system

Harain committed:
Revert "remove vanilla _LaR_units_infantry.asset"

This reverts commit 60dfcd139b7883e807fe4954eeecb0074238ea2b.

Harain commented:
fixed over 2k duplicates, only 8 left which for some reason are still crating a duplicate error even tho in replace
moved them all to localisation\english\replace\r56_replace2_l_english.yml to keep separate

tried to keep criteria similar, but it's hard with so many:
- deleted exact vanilla copies (unless close to a desc where I left it commented
- lots of duplicates in same file deleted or commented if unsure of which text was better
- when deleting duplicates, always kept the ones either different from vanilla or most recent
- fixed dozens of locs that were missing _desc or something similar, same for some event locs
- i'm probably forgetting other special stuff, but it's all in the commit descriptions

marijn211 reviewed (changes requested):
Since this is based on the bugfix branch, and contains quite a lot of changes, I will check this after accepting the bugfixing branch

marijn211 committed:
Merge branch 'Beta-Edits' into loc-key-collisions

SpicyAlfredo commented:
> Since this is based on the bugfix branch, and contains quite a lot of changes, I will check this after accepting the bugfixing branch

Eyeballing it, everything looks right. If I don't have time to check before the first this might just be a 'fuck we ball' merge and we fix quirks with it before the 7th

marijn211 reviewed (changes requested):
By moving parties_l_english.yml out of the replace folder (which in principle is a good idea), we stop inheriting lines that are only in the vanilla file of that name, some of which seem to be unported

Harain committed:
rework parties_l_english.yml

renamed it to replaced_parties_l_english.yml and moved to replace
this means we will now use the vanilla parties_l_english.yml, with our custom locs in replaced_parties_l_english.yml
many there are still same as vanilla, too many for me to check individually at this point
there are now replaced party locs in 3 files... sorry
- replaced_parties_l_english.yml
- r56_replace_l_english.yml - the file I made specifically for this PR, stuff there can be moved to proper place if anyone cares to do that
- r56_parties_l_english.yml - this file was already there and I just noticed now lol... sorry again

also 3 more duplicate locs even in replace, can't understand this
added to r56_replace2_l_english.yml

marijn211 committed:
Merge branch 'Beta-Edits' into loc-key-collisions

marijn211 reviewed (approved):
Double checking rn but looks good
